### PR TITLE
Namespace is now a flag for all the commands

### PR DESCRIPTION
**Pull request by juanfont**
This PR changes how namespaces are specified in the commands of Headscale. Basically mimicking how `kubectl` behaves.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ Suggestions/PRs welcomed!
 
 8. In the server, register your machine to a namespace with the CLI
   ```shell
-  ./headscale register YOURMACHINEKEY myfirstnamespace
+  ./headscale -n myfirstnamespace register YOURMACHINEKEY
```
Comment (cure) [inline README.md:81]: There seems to be a trailing space on this line.

File: cmd/headscale/headscale.go [hunk 2]
```diff
@@ -83,21 +83,35 @@ func main() {
...
 	headscaleCmd.AddCommand(cli.NamespaceCmd)
 	headscaleCmd.AddCommand(cli.NodeCmd)
+	headscaleCmd.AddCommand(cli.PreauthkeysCmd)
+	headscaleCmd.AddCommand(cli.RegisterCmd)
+	headscaleCmd.AddCommand(cli.RoutesCmd)
+	headscaleCmd.AddCommand(cli.ServeCmd)
+	headscaleCmd.AddCommand(versionCmd)
+
+	cli.NodeCmd.PersistentFlags().StringP("namespace", "n", "", "Namespace")
+	cli.NodeCmd.MarkPersistentFlagRequired("namespace")
```
Comment (cure) [inline cmd/headscale/headscale.go:95]: Can we check for any returned errors here? Also on line 98, 101, 104.

**Review by cure — APPROVED** [review]
Thanks! Just a few small things.

**Comment by juanfont** [comment]
Thanks for the heads up!

**Pull request merged**